test.ts

```diff
@@ -16,18 +17,17 @@ describe('update', () => {
       expect(update([1], {$push: [7]})).toEqual([1, 7]);
     });
     it('does not mutate the original object', () => {
-      const obj = [1];
-      update(obj, {$push: [7]});
-      expect(obj).toEqual([1]);
+      const obj = Object.freeze([1]);
+      expect(() => update(obj, {$push: [7]})).toNotThrow();
     });
     it('only pushes an array', () => {
-      expect(update.bind(null, [], {$push: 7})).toThrow(
```

Comment:
When `use strict` is enabled it's not happy with `null` as the lexical `this` context.